pkgs/development/tools/misc/ccache/default.nix

```diff
@@ -15,43 +15,47 @@ let ccache = stdenv.mkDerivation rec {
     substituteInPlace Makefile.in --replace 'objs) $(extra_libs)' 'objs)'
   '';
 
-  doCheck = true;
+  doCheck = !stdenv.isDarwin;
 
-  passthru = {
+  passthru = let
+      cc = stdenv.cc.cc;
+      ccname = if stdenv.cc.cc.isClang then "clang" else "gcc";
+      cxxname = if stdenv.cc.cc.isClang then "clang++" else "g++";
```

Comment:
It might not work very well in cases when build scripts do different things depending on compiler's name.